WDL/__init__.py

```diff
@@ -52,13 +51,78 @@ def load(
         uri,
         path=path,
         check_quant=check_quant,
-        import_uri=import_uri,
+        read_source=read_source,
         import_max_depth=import_max_depth,
-        source_text=source_text,
     )
 
 
+async def load_async(
+    uri: str,
+    path: Optional[List[str]] = None,
+    check_quant: bool = True,
+    read_source: Optional[
+        Callable[[str, List[str], Optional[Document]], Awaitable["ReadSourceResult"]]
+    ] = None,
+    import_max_depth: int = 10,
+) -> Document:
+    """
+    Async version of :func:`load`, with all the same arguments
+    """
+    return await Tree.load_async(
+        uri,
+        path=path,
+        check_quant=check_quant,
+        read_source=read_source,
+        import_max_depth=import_max_depth,
+    )
+
+
+async def read_source_default(
+    uri: str, path: List[str], importer: Optional[Document]
+) -> "ReadSourceResult":
+    """
+    Default async routine for the ``read_source`` parameter to :func:`load` and :func:`load_async`,
+    which they use to read the desired WDL document and its imports. This default routine handles
+    local files only, supplying the search path logic to resolve relative filenames; it fails with
+    network URIs.
+
+    :param uri: Filename/URI to read, as provided to :func:`load` or the WDL import statement; may
+                be relative
+    :param path: Local directories to search for relative imports
+    :param importer: The document importing the one here requested, if any; the
+                     ``importer.pos.uri`` and ``importer.pos.abspath`` fields may be relevant to
+                     resolve relative imports.
+    :returns: ``ReadSourceResult(source_text="...", abspath="...")``
+
+    Callers may wish to override ``read_source`` with logic to download source code from network
+    URIs, and for local filenames fall back to ``return await WDL.read_source_default(...)``.
+
+    Note: the synchronous :func:`load` merely calls :func:`load_async` on the current
+    ``asyncio.get_event_loop()`` and awaits the result.
+    """
+    return await Tree.read_source_default(uri, path, importer)
+
+
+class ReadSourceResult(Tree.ReadSourceResult):
+    """
+    The ``NamedTuple`` to be returned by the ``read_source`` routine. Its ``source_text: str`` field
+    provides the WDL source code, and the ``abspath: str`` field is the absolute filename/URI from
+    which the source was read (e.g. after resolving a relative path).
+    """
+
+
+async def resolve_file_import(uri: str, path: List[str], importer: Optional[Document]) -> str:
```

Comment:
@dinvlad ok, sold :) I made `read_source_default` call this, rather than taking its result as an argument, in order to avoid having to make them both overridable `load()` arguments

Reply:
Perfect, thanks!!